### ISSUE TITLE
Property enumeration segfault

### ISSUE THREAD

**Issue by zherczeg** [pando-project/escargot#91]
```
var props = {
  get bar() {
    delete props.prop2;
    return { value : 2, writable : true };
  },
  prop2: {
    value: 3,
    writable: true,
  },
};

Object.defineProperties({}, props);
```

It seems the problem is that it goes through the properties of a structure, while the structure is changed in the meantime.

**Comment by pmarkee** [comment]
I think this was fixed by #209 , I just checked on the latest master and there was no segfault.

**Comment by renatahodovan** [comment]
I cannot repro this one either on 958b293. However, a probably related issue #135 still crashes.
